singer-io/tap-recharge
# v.2.0.0 Major revision to use updated Recharge API Version

# Description of change
Major revision to use new version of the Recharge-API [2021-11](https://docs.rechargepayments.com/changelog/2021-11-api-release-notes). This deprecates `collections`, adds `payment_methods`, changes `shop` endpoint to `store`, and adds/removes fields in all other objects. Also, added retry to `client.py` and decrease batch size to address [intermittent unterminated string issue](https://github.com/singer-io/tap-recharge/issues/4).

# Manual QA steps
 Tested all endpoints with singer-check-tap (singer-tools), replication to target-stitch (with and without previous state), and verified pylint code quality.
 
# Risks
 Major version release: adds 1 new endpoint, deprecates 1 endpoint, and adds/removes fields to all other endpoints. Recharge offers different client configurations which enables/disables some endpoints. Added a discovery/schema check to only allow available endpoints.
 
# Rollback steps
Revert to previous version: v.1.1.2
